packages/pg-pubsub/src/PgSubscriptionResolverPlugin.ts

```diff
@@ -4,10 +4,42 @@ import { PubSub, withFilter } from "graphql-subscriptions";
 
 const debug = debugFactory("pg-pubsub");
 
-function isPubSub(pubsub: any): pubsub is PubSub {
+function isPubSub(pubsub: unknown): pubsub is PubSub {
   return !!pubsub;
 }
 
+function withInitialEvent(
+  topic: string,
+  initialEvent: (
+    args: { [key: string]: unknown },
+    resolveContext: unknown,
+    resolveInfo: unknown
+  ) => unknown
+) {
+  return async function* subscribeWithInitialEvent(
+    asyncIterator: AsyncIterator<unknown>,
+    _parent: unknown,
+    args: { [key: string]: unknown },
+    resolveContext: unknown,
+    resolveInfo: unknown
+  ) {
+    const event = await initialEvent(args, resolveContext, resolveInfo);
+    if (event != null) {
+      if (typeof event !== "object") {
+        throw new Error(
+          "event returned from pgSubscription initialEvent must be an object"
+        );
+      }
+      yield { ...event, topic };
+    }
+
+    // @ts-ignore code is actually fine, but async iterators are currently an ES.next feature
```

Comment:
Seems it requires Node 10.3.0+ but we need to support Node 8.6. May need to write a heavier wrapper :disappointed:

You can leave this with me if you're not in a rush (I think you have your own fork?)

Reply:
No rush. Do you need anything else from me? Access to the forked repo?